frontend/src/components/Profile/Profile.tsx

```diff
@@ -9,19 +9,35 @@ import {
 } from '@mui/material';
 import { AccountCircle } from '@mui/icons-material';
 import ComponentContainer from '../common/ComponentContainer';
+import useUser from '@/hooks/useUser';
+import { useClient } from '@/contexts/AppContext';
 
 const InnerProfile: React.FC = () => {
-  const [formData, setFormData] = useState({
+  type UserMetadata = {
+    name: string;
+    email: string;
+    phone: string;
+    address: string;
+    company: string;
+    description: string;
+  };
+
+  const [isEditing, setIsEditing] = useState(false);
+  const user = useUser();
+  const client = useClient();
+
+  const defaultFormData: UserMetadata = {
     name: '',
     email: '',
     phone: '',
     address: '',
     company: '',
     description: '',
-  });
+  };
 
   const [open, setOpen] = useState(false);
   const [image, setImage] = useState<string | null>(null);
+  const [formData, setFormData] = useState<UserMetadata>(defaultFormData);
```

Comment:
Instead of `defaultFormData`, we can put `user.metadata` right?

Reply:
In fact, you can just put `{ ...user.meta_data, email: user.email as string }` right?

Reply:
That way we can get rid of the useEffect on line 51. And less code === better code.

Reply:
I have the defaultFormData for the unlogged in state,cuz as we logged out, it wont turn to Home page but stay the profile. So when the user is logged out, it will show empty state.

Reply:
Ah yes. I think you can remove it still, we will do another card/PR for making sure that if a page requires a user to be logged in, they will be required to login first, then go to the page.

Reply:
Yeah, we can separate the Check user logged for each page. Will remove the default form data

Reply:
![image](https://github.com/mursal-ashraf/bankan/assets/100513480/b823a35d-644f-4164-b87d-61dd0682e900)
Edit the default Form Date